src/tools/rustfmt/src/macros.rs

```diff
@@ -1237,10 +1237,10 @@ impl<'a> MacroParser<'a> {
                 span,
             },
             _,
-        )) = self.toks.look_ahead(0)
+        )) = self.iter.peek()
```

Comment:
I see that we're replacing calls to `.look_ahead()` with `.peek()`. Is `.peek()` equivalent to both `.look_ahead(0)` here and `.look_ahead(1)` above in `MacroParser::parse`?

Reply:
Yes. Full details are in the individual commit messages.

Reply:
Interesting that it doesn't matter if you use `0`, `1`, or `2` as the look ahead, and that things only start failing if you choose `3` or more. I know you mentioned it's due to the structure of the code, but I'm wondering if you could elaborate a little more to help me fully understand.

Reply:
AIUI, the existing code says "If there are at least two tokens remaining, try to parse a branch". The new code says "if there is at least one token remaining, try to parse a branch". `parse_branch` has three `self.toks.next()?` calls in it, and all three must succeed for the function to succeed, i.e. there must actually be at least three tokens remaining.